nodes/main/src/command.rs

```diff
@@ -165,7 +191,12 @@ pub fn run() -> Result<()> {
                     backend,
                     ..
                 } = new_partial(&config)?;
-                Ok((cmd.run(client, backend), task_manager))
+                let aux_revert = Box::new(move |client: Arc<FullClient>, backend, blocks| {
+                    sc_consensus_babe::revert(client.clone(), backend, blocks)?;
+                    grandpa::revert(client, blocks)?;
```

Comment:
It seems possible that BABE could fail to revert and in that state, GRANDPA would never be called. If this is acceptable, proceed; but I'm personally not certain if the chain state would be acceptable going forward.

Reply:
This is a true concern. However, I would rather keep the code here the same as in upstream, and raise these concerns on Substrate first.

Reply:
agree with @aliXsed  suggestion